driver-core/src/main/java/com/datastax/driver/core/BytesToSegmentDecoder.java

```diff
@@ -52,11 +52,18 @@
 
   @Override
   protected Object decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception {
-    ByteBuf payloadAndCrc = (ByteBuf) super.decode(ctx, in);
-    assert header != null;
+    ByteBuf payloadAndCrc;
+    try {
+      payloadAndCrc = (ByteBuf) super.decode(ctx, in);
+    } catch (Exception e) {
+      // Don't hold on to a stale header if the parent method discarded the "frame"
+      header = null;
```

Comment:
Actually this doesn't even matter currently, because we always drop the connection when an error happens here.
But just being safe in case that ever changes.